elastic/kibana
# [Security Solution][Entity Analytics] More detailed privilege control for Asset Criticality assignment

## Summary

This PR fixes a bug where users with `read` access and without `write` access would not be able to see the Criticality assigned to an entity.
The criticality component would only check whether a user has _all_ the required privileges, which didn't account for read-only case. 

By adding more granular and detailed information to the privileges API call, we can now show the currently assigned criticality but hide the `Change/Create` button if the user does not have `write` access.

https://github.com/elastic/kibana/assets/2423976/00845ac7-11f4-429c-990b-c068dfb19f6b



### How to test

1. Login in as a `superuser`
2. Assign criticality to some entity via the expandable flyout.
3. Create a role with read-only/read-write or no access to the criticality index

<img width="1352" alt="Screenshot 2024-01-31 at 09 58 04" src="https://github.com/elastic/security-team/assets/7609147/22477433-63e8-4ccf-a077-275725a600aa">

4. Create a user(s) with the new role(s)
5. Login with that user and open the expandable flyout
6. Check that the criticality is only shown for users with `read` privilege and that the update button only shows for users with `write` privileges







### Checklist

Delete any items that are not applicable to this PR.
